### PR TITLE
Skip automatic checkpoint output on time step 0, catch up later

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #24252

### CODE REVIEW

**Review by roystgnr — APPROVED** [review]

File: framework/src/outputs/Checkpoint.C [hunk 1]
```diff
@@ -104,21 +104,26 @@ Checkpoint::shouldOutput(const ExecFlagType & type)
...
   // as the autosave and that checkpoint isn't on its interval, then output.
-  if (_is_autosave == SYSTEM_AUTOSAVE || (_is_autosave == MODIFIED_EXISTING && !shouldOutput))
+  if (_is_autosave == SYSTEM_AUTOSAVE || (_is_autosave == MODIFIED_EXISTING && !should_output))
   {
     // If this is a pure system-created autosave through AutoCheckpointAction,
     // then sync across processes and only output one time per signal received.
     comm().max(Moose::interrupt_signal_number);
-    shouldOutput = (Moose::interrupt_signal_number != 0);
-    if (shouldOutput)
+    // Reading checkpoint on time step 0 is not supported
```
Comment (loganharbour) [inline framework/src/outputs/Checkpoint.C:117]: We're imposing a limitation on production code that is needed for test code, which I don't think is a good one. IF someone wants to send a signal as soon as a run starts, as it's setting up, to get a checkpoint immediately.. they should be able to. Signaled checkpoints should always be the _next possible_ checkpoint, it doesn't make sense to do anything else. I didn't have this opinion before, but I feel strongly about it now

Comment (GiudGiud) [reply framework/src/outputs/Checkpoint.C:117]: while in terms of "why" this PR was made you are right that we are doing it for testing purposes, the limitation here is that we cannot read time step 0 checkpoints, both in testing and production code.

So the fix you want would be to enable this. I can take care of this if I rework the Checkpoint system. I'm not working on this until I am done with the Positions project.

Comment (roystgnr) [reply framework/src/outputs/Checkpoint.C:117]: Logan managed to convince me of this too, in Slack:

--

> let’s say you have… four rounds of adaptivity with file reader mesh, on 20 nodes, with a image reader,  and that run takes 3 hours (which I have worked with someone on..)

Dear God, 3 hours for initial adaptivity?  And that wasn't a bug?

> there’s multiapp solves on initial
> special problem with porous zones

Okay, I'm wrong then.
I hadn't even considered the possibility of solves before you solve.

--

This seemed like a decent temporary workaround that wouldn't actually impact production cases, but I guess not so much.

Comment (loganharbour) [reply framework/src/outputs/Checkpoint.C:117]:
```
[Outputs/checkpoint]
    type = Checkpoint
    execute_on = 'INITIAL
[]
```

is supported, and you can `Problem/restart_file_base` from this

**Pull request merged**